interview-questions.md

```diff
@@ -9,59 +9,63 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
-
-  Researched answer:
-
+  Your answer: 'this' is the means of letting your computer know that the variable or function you want to call on is local to the current code.
 
+  Researched answer: 'this' is a keyword used to reference an object via use of dot notation. Its purpose is to tell a program where to access reference values or data from.
+  
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is a coding module that uses JSX, a language that allows for creative representation of javascript through use of integrated HTML. Its purpose is to allow for dynamic pages that have extremely fast load times due to it only refreshing the part of the page/script where a change is being made. I would use it for pages/sites that are intended to be interactive, dynamic, with little rendering time.
 
-  Researched answer:
+  Researched answer: React is a code library used to build interactive user interfaces. HTML and CSS are used curate the visual side of an application, while JS is used to create functionality. React is useful because it efficiently updates and renders on the components that have undergone change, resulting in greater speed and efficiency than other means of coding.
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: render() is required; it acts as the javascript command to "run" the JSX that is within its curly braces.
 
-  Researched answer:
+  Researched answer: render() is the only required method in a React class. When it is called it is seeking this.props and/or this.state, then returning one of the following: react elements ( <Component /> ), arrays and fragments, portals, string and numbers, or booleans or null. The render() function does not modify components, returns the same result per each invocation, and does not directly interact with the browser.
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: JSX is a language that allows for creative representation of javascript through use of integrated HTML.
 
+  Researched answer: JSX (JavaScript Syntax Extension) is syntatic sugar that produces elements in React. Although use of JSX is not required in React it is useful because it allows programmers to work on visual and functional code simultaneously, and it allows for more useful error messages. The major difference between JSX and HTML is that JSX requires a return, or it won't compile.
```

Comment:
Awesome answer! pad out your time in interviews by sharing a time you used a tool or technology and explain how you overcame a blocker with it. Should buy you a few minutes and you can more easily connect with your interviewer.